app/code/community/Hps/Securesubmit/Model/Payment.php

```diff
@@ -252,6 +243,29 @@ private function _authorize(Varien_Object $payment, $amount, $capture)
         return $this;
     }
 
+    /**
+     * @param Varien_Object|Mage_Sales_Model_Order_Payment  $payment
+     * @param float                                         $amount
+     * @param HpsReportTransactionDetails|HpsAuthorization  $response
+     * @param Mage_Payment_Model_Method_Abstract::STATUS_UNKNOWN|STATUS_APPROVED|STATUS_ERROR|STATUS_DECLINED|STATUS_VOID|STATUS_SUCCESS                                   $status
+     */
+    protected function closeTransaction($payment, $amount, $response, $status = self::STATUS_APPROVED){
+        $payment->setStatus($status);
+        $payment->setAmount($amount);
+        if (property_exists($response,'cardType')){
+            $payment->setCcType($response->cardType);
```

Comment:
It actually seems to ignore the card type unless you process it in the admin panel in which case there is an encrypted card number stored. the type seems to depend then on the decrypted card number
I have changed it to the card type in the payments object as suggested